dali/test/python/test_eager_coverage.py

```diff
@@ -0,0 +1,1022 @@
+    pipe = pipe_fun(fn_op, kwargs, source=fn_source, layout=layout)
+
+    compare_eager_with_pipeline(pipe, eager_op, eager_source=eager_source, layout=layout, **kwargs)
+
+
+@pipeline_def(batch_size=batch_size, num_threads=4, device_id=None)
+def no_input_pipeline(op, kwargs):
+    out = op(**kwargs)
+    if isinstance(out, list):
+        out = tuple(out)
+    return out
+
+
+def check_no_input(op_path, *, fn_op=None, eager_op=None, batch_size=batch_size, N_iterations=5,
+                   **kwargs):
+    fn_op, eager_op = get_ops(op_path, fn_op, eager_op)
+    pipe = no_input_pipeline(fn_op, kwargs)
+    pipe.build()
+
+    for _ in range(N_iterations):
+        out_fn = pipe.run()
+        out_eager = eager_op(batch_size=batch_size, **kwargs)
+
+        if not isinstance(out_eager, (tuple, list)):
+            out_eager = (out_eager,)
+
+        assert len(out_fn) == len(out_eager)
+
+        for tensor_out_fn, tensor_out_eager in zip(out_fn, out_eager):
+            assert type(tensor_out_fn) == type(tensor_out_eager)
+            check_batch(tensor_out_fn, tensor_out_eager, batch_size)
+
+
+@pipeline_def(batch_size=batch_size, num_threads=4, device_id=None)
+def reader_pipeline(op, kwargs):
+    out = op(pad_last_batch=True, **kwargs)
+    if isinstance(out, list):
+        out = tuple(out)
+    return out
+
+
+def check_reader(op_path, *, fn_op=None, eager_op=None, batch_size=batch_size, N_iterations=2,
+                 **kwargs):
+    fn_op, eager_op = get_ops(op_path, fn_op, eager_op)
+    pipe = reader_pipeline(fn_op, kwargs)
+    pipe.build()
+
+    iter_eager = eager_op(batch_size=batch_size, **kwargs)
+
+    for _ in range(N_iterations):
+        for i, out_eager in enumerate(iter_eager):
+            out_fn = pipe.run()
+
+            if not isinstance(out_eager, (tuple, list)):
+                out_eager = (out_eager,)
+
+            assert len(out_fn) == len(out_eager)
+
+            for tensor_out_fn, tensor_out_eager in zip(out_fn, out_eager):
+                if i == len(iter_eager) - 1:
+                    tensor_out_fn = _slice_tensorlist(tensor_out_fn, len(tensor_out_eager))
+
+                assert type(tensor_out_fn) == type(tensor_out_eager)
+                check_batch(tensor_out_fn, tensor_out_eager, len(tensor_out_eager))
+
+
+@pipeline_def(batch_size=batch_size, num_threads=4, device_id=None)
+def file_reader_pipeline(kwargs):
+    data, _ = fn.readers.file(**kwargs)
+    return data
+
+
+@pipeline_def(batch_size=batch_size, num_threads=4, device_id=None)
+def reader_op_pipeline(op, kwargs, source=None, layout=None):
+    if source is None:
+        raise RuntimeError('No source for file reader.')
+    data, _ = fn.readers.file(file_root=source)
+    out = op(data, **kwargs)
+    if isinstance(out, list):
+        out = tuple(out)
+    return out
+
+
+def test_image_decoder():
+    check_single_input('decoders.image', pipe_fun=reader_op_pipeline, fn_source=images_dir,
```

Comment:
I used the generic one as much as possible to save time and lines of code. But some were easier to write and in my opinion to read (when they are a bit more complicated) as individual pipelines. Additionally I think in the future when someone will be writing tests for their new operator they will most likely write their own pipelines, so it will come up anyway.